core/tx_pool.go

```diff
@@ -1162,3 +1159,68 @@ func (as *accountSet) containsTx(tx *types.Transaction) bool {
 func (as *accountSet) add(addr common.Address) {
 	as.accounts[addr] = struct{}{}
 }
+
+// txLookup is used to track all transactions to allow lookups without contention
+type txLookup struct {
+	all  map[common.Hash]*types.Transaction
+	lock sync.RWMutex
+}
+
+func newTxLookup() *txLookup {
+	return &txLookup{
+		all: make(map[common.Hash]*types.Transaction),
+	}
+}
+
+// calls f on each key and value present in the map
```

Comment:
Please add the method name too when documenting, that's the general Go style. Eg.

`// Range calls f on each key and value present in the map.

Reply:
(Please do this and the punctuation mark for every method).